src/ai/ai.cpp

```diff
@@ -346,7 +348,15 @@ void ShipAI::runOrders()
                 flyTowards(owner->getOrderTargetLocation());
             }
         }else{
```

Comment:
I've encountered this case, Roaming mode action condition and prioritization should be changed.